core/tests/webdriverio_utils/ContributorDashboardPage.js

```diff
@@ -70,7 +70,7 @@ var ContributorDashboardPage = function() {
     await _openLanguageSelector();
     var selectorOption = selectorContainer.$(
       `.e2e-test-language-selector-option=${language}`);
-    await action.click(`Language ${language} option`, selectorOption);
+    await action.select(`Language ${language} option`, selectorOption);
```

Comment:
I think you won't have to use `_openLanguageSelector` as select will open the selector and click the given option.

Also, you'll have to pass 3 parama: https://github.com/oppia/oppia/blob/0e13814a67ae59d52acc31a8694abcedde8937fa/core/tests/webdriverio_utils/action.js#L70

Reply:
I think select will not open the selector, we have to open it prior to calling action.select.

Reply:
Using action.select it throws up error of `Cannot convert object to primitive value` 

Changed code:-
  var _selectLanguage = async function(language) {
    var selectorOption = selectorContainer.$(
      `.e2e-test-language-selector-option=${language}`);
    await action.select(
      `Language ${language} option`, selectorContainer, selectorOption);
  };

Reply:
Hi @DubeySandeep, thanks for the suggestion but `action.select` is not able to detect the option in this scenario.

It throws the error `Option with text "shqip (Albanian)" not found.`

Reply:
I would suggest reading the action.select method and check how it works. I think you have to pass the selector element and not the option element. 

Yes, we want all the user actions to be performed via (action.*) method as the action.* methods are properly guarded to avoid flakiness.

Reply:
I found another method in action.js: matSelect, maybe you can use this: https://github.com/oppia/oppia/blob/0e13814a67ae59d52acc31a8694abcedde8937fa/core/tests/webdriverio_utils/action.js#L78

(Make sure to read the code and usage before applying it. I hope this helps! :))

Reply:
Actually my question was `click` is also a action method so any specific reason behind not using click and using select instead? 

I'll try out your suggestions and report back to you. Thanks.

Reply:
Hi @DubeySandeep on researching a bit more, I found following conclusions please let me know about your thoughts on this.

- `action.select` calls `selectByVisibleText` internally
- `selectByVisibleText` according to officially documentation works on `select` component.
- In our case here, we are not using select rather we are rendering div in a for loop. The code we are testing can be found [here](https://github.com/oppia/oppia/blob/develop/core/templates/pages/contributor-dashboard-page/translation-language-selector/review-translation-language-selector.component.html#L21).
- The previous code was using `click` which is also a _action._ method so I think we can use that.

Reply:
> Actually my question was click is also a action method so any specific reason behind not using click and using select instead?

The reason behind using select is that we want to perform users' UI interactions using action.js methods and `select from a dropdown` is the user interacting with UI. Though the overall journey requires click, we want to pack the overall journey in a reusable code for consistency and avoid introducing different processes for `select` in different parts of the test, helping us avoid/control flakiness introduced via an unexpected pattern for `select` workflow.  I hope this explains, let me know if you have any other thoughts/idea.

Reply:
> action.select calls selectByVisibleText internally

Have you tried aaction.matSelect? (I think you are using `mat-select` component in the UI, right?`

Reply:
Hi @DubeySandeep thanks for the clarfication, I totally get your point.

Actually the code is not using `select/mat-select`. It is just simple div and in this case I fear we can't use `action.select`. You can see the UI code [here](https://github.com/oppia/oppia/blob/develop/core/templates/pages/contributor-dashboard-page/translation-language-selector/review-translation-language-selector.component.html#L21).

Reply:
Why aren't we using select/mat-select (provides better a11y feature)? Any specific reason? 

Also, a few more questions:

1. Why this file/component is placed inside `core/templates/pages/contributor-dashboard-page/translation-language-selector/` and not into it's own dir?
2. Why can't we use  `translation-language-selector` component?

Reply:
Hi @DubeySandeep I understand your concerns but this code is not introduced by me rather I am using it for setup steps to populate the stats which I need to test my feature,

1. On researching a bit these files were introduced in PR https://github.com/oppia/oppia/pull/16157. 
2. `translation-language-selector` is available is translate text tab in contributor dashboard but on `My contribution` tab we have `review-translation-language-selector`.